### PR TITLE
Extends controller-runtime with new metric

### PR DESCRIPTION
**Pull request by alexandrealvino**
Extends controller runtime metrics adding prometheus metrics to count cluster static entries failures

Fix issue #278

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -42,11 +42,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	k8sMetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
```
Comment (faisal-memon) [inline cmd/main.go:45]:
```suggestion
	k8smetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
```

File: cmd/main.go [hunk 2]
```diff
@@ -78,6 +80,10 @@ func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
 	utilruntime.Must(spirev1alpha1.AddToScheme(scheme))
+
+	k8sMetrics.Registry.MustRegister(
```
Comment (faisal-memon) [inline cmd/main.go:84]:
```suggestion
	k8smetrics.Registry.MustRegister(
```

**Review by azdagron — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -9,7 +9,8 @@ require (
 	github.com/jpillora/backoff v1.0.0
 	github.com/onsi/ginkgo/v2 v2.20.2
 	github.com/onsi/gomega v1.34.2
-	github.com/spiffe/go-spiffe/v2 v2.3.0
+	github.com/prometheus/client_golang v1.19.0
+  github.com/spiffe/go-spiffe/v2 v2.3.0
```
Comment (kfox1111) [inline go.mod:13]:
```suggestion
	github.com/prometheus/client_golang v1.19.1
	github.com/spiffe/go-spiffe/v2 v2.3.0
```

**Comment by azdagron** [comment]
Thanks for testing this IRL, @kfox1111!

**Review by azdagron — APPROVED** [review]

**Pull request merged**